packages/server/lib/browsers/browser-cri-client.ts

```diff
@@ -255,22 +267,111 @@ export class BrowserCriClient {
   }
 
   static async _onAttachToTarget (options: AttachedToTargetOptions) {
-    const { browserClient, event, protocolManager } = options
+    const { browserClient, browserCriClient, CriConstructor, event, host, port, protocolManager } = options
+    const CreateCRI = CriConstructor || CRI
+    const { sessionId, targetInfo, waitingForDebugger } = event
+    let { targetId, url } = targetInfo
 
-    debug('Target.attachedToTarget %o', event.targetInfo)
+    debug('Target.attachedToTarget %o', targetInfo)
 
     try {
-      if (event.targetInfo.type !== 'page') {
+      // The basic approach here is we attach to targets and enable network traffic
+      // We must attach in a paused state so that we can enable network traffic before the target starts running.
+      if (targetInfo.type !== 'page') {
         await browserClient.send('Network.enable', protocolManager?.networkEnableOptions ?? DEFAULT_NETWORK_ENABLE_OPTIONS, event.sessionId)
       }
+    } catch (error) {
+      // it's possible that the target was closed before we could enable
+      // network and continue, in that case, just ignore
+      debug('error running Network.enable:', error)
+    }
+
+    if (!waitingForDebugger) {
+      debug('Not waiting for debugger (id: %s)', targetId)
+
+      // targets created before we started listening won't be waiting
+      // for debugger and are not extras
+      return
+    }
 
-      if (event.waitingForDebugger) {
-        await browserClient.send('Runtime.runIfWaitingForDebugger', undefined, event.sessionId)
+    async function run () {
+      try {
+        await browserClient.send('Runtime.runIfWaitingForDebugger', undefined, sessionId)
+      } catch (error) {
+        // it's possible that the target was closed before we could enable
+        // network and continue, in that case, just ignore
+        debug('error running Runtime.runIfWaitingForDebugger:', error)
       }
-    } catch (error) {
-      // it's possible that the target was closed before we could enable network and continue, in that case, just ignore
-      debug('error attaching to target browser', error)
     }
+
+    // the url often isn't specified with this event, so we get it
+    // from Target.getTargets
+    if (!url) {
+      const { targetInfos } = await browserClient.send('Target.getTargets')
+
+      const thisTarget = targetInfos.find((target) => target.targetId === targetId)
+
+      if (thisTarget) {
+        url = thisTarget.url
```

Comment:
Out of curiosity, what does the url show as for service workers/web workers/shared workers? We want to make sure we're still using the proxy for them for workers started in the main target.

Can maybe use the system tests here to see? 

https://github.com/cypress-io/cypress/blob/develop/system-tests/test/web_worker_spec.js
https://github.com/cypress-io/cypress/blob/develop/system-tests/test/service_worker_spec.js

Reply:
For service workers, we can go off `targetInfo.type`. I think we only want to consider `type === 'page'` as an extra target, so I [updated the logic here](https://github.com/cypress-io/cypress/pull/28188/commits/5f72ec9f7f1ec10105fc3ff31994f1184c6b3fd8). Before that change, those system tests were failing, so looks like they were doing their jobs.